app/scripts/migrations/157.ts

```diff
@@ -0,0 +1,41 @@
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 157;
+
+/**
+ * This migration deletes the preference `disabledUpgradeAccountsByChain` if the user has
+ * existing data.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly
+ * what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by
+ * controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  const preferencesControllerState = state?.PreferencesController as
+    | Record<string, unknown>
+    | undefined;
+
+  const preferences = preferencesControllerState?.preferences as
+    | Record<string, unknown>
+    | undefined;
+
+  delete preferences?.disabledUpgradeAccountsByChain;
```

Comment:
The migration should also remove `accountUpgradeDisabledChains`. That is what is being persisted in the state on 12.17.X and there was no migration to remove it on the [PR that introduced the `disabledUpgradeAccountsByChain`](https://github.com/MetaMask/metamask-extension/pull/31774)

Reply:
Good point, let me update the PR.

Reply:
I updated the PR